### PR TITLE
interop-test: port metadata_filter_test case to x8s 

### PR DESCRIPTION
**Pull request by YifeiZhuang**
Similar to https://github.com/grpc/grpc/pull/27534, did simplification compared to original GCE test case.
TODO: kokoro run
Only merge after current interop-tests are stable

cc @sergiitk

### CODE REVIEW

File: tools/run_tests/xds_k8s_test_driver/tests/url_map/metadata_filter_test.py [hunk 1]
```diff
@@ -0,0 +1,222 @@
...
+                }]
+            }],
+            'service': GcpResourceManager().alternative_backend_service()
+        }]
+        return host_rule, path_matcher
+
+    def xds_config_validate(self, xds_config: DumpedXdsConfig):
+        self.assertNumEndpoints(xds_config, 2)
+        self.assertEqual(len(xds_config.rds['virtualHosts'][0]['routes']), 2)
+        logger.info('config:%s', xds_config)
```
Comment (lidizheng) [inline tools/run_tests/xds_k8s_test_driver/tests/url_map/metadata_filter_test.py:75]: As discussed offline, let's try to be more concrete about if we are receiving the correct xDS config.

File: tools/run_tests/xds_k8s_test_driver/tests/url_map/metadata_filter_test.py [hunk 2]
```diff
@@ -0,0 +1,222 @@
...
+DumpedXdsConfig = xds_url_map_testcase.DumpedXdsConfig
+RpcTypeUnaryCall = xds_url_map_testcase.RpcTypeUnaryCall
+RpcTypeEmptyCall = xds_url_map_testcase.RpcTypeEmptyCall
+XdsTestClient = client_app.XdsTestClient
+
+logger = logging.getLogger(__name__)
+flags.adopt_module_key_flags(xds_url_map_testcase)
+
+_NUM_RPCS = 150
+match_labels = [{'name': 'TRAFFICDIRECTOR_NETWORK_NAME', 'value': 'default'}]
```
Comment (lidizheng) [inline tools/run_tests/xds_k8s_test_driver/tests/url_map/metadata_filter_test.py:37]: Let's use "default-vpc".

File: tools/run_tests/xds_k8s_test_driver/tests/url_map/metadata_filter_test.py (hunk 1 above)
Comment (YifeiZhuang) [reply tools/run_tests/xds_k8s_test_driver/tests/url_map/metadata_filter_test.py:90]: Thanks Lidi. Updated.

Once approved we will run this before merge:
```
python3 -m tests.url_map --flagfile=config/grpc-testing.cfg --kube_context=gke_grpc-testing_us-central1-c_interop-test-psm-basic --client_image=gcr.io/grpc-testing/xds-interop/cpp-client:a069a322a9b3f49f0c2bdbd5ee363bfbab2471b0 --testing_version=master  --flagfile=config/url-map.cfg --strategy create
```
correct?

File: tools/run_tests/xds_k8s_test_driver/tests/url_map/metadata_filter_test.py [hunk 3]
```diff
@@ -0,0 +1,270 @@
...
+        self.assertEqual(
+            xds_config.rds['virtualHosts'][0]['routes'][0]['match']['headers']
+            [0]['exactMatch'], _TEST_METADATA_VALUE_EMPTY)
+        self.assertEqual(
+            xds_config.rds['virtualHosts'][0]['routes'][1]['match']['prefix'],
+            "/grpc.testing.TestService/Unary")
+        self.assertEqual(
+            xds_config.rds['virtualHosts'][0]['routes'][2]['match']['prefix'],
+            "")
+        logger.info('config:%s', xds_config)
```
Comment (lidizheng) [inline tools/run_tests/xds_k8s_test_driver/tests/url_map/metadata_filter_test.py:257]: The config will be dumped if this test case failed. If we log at every attempt, the driver log can be quite verbose.

**Review by lidizheng — APPROVED** [review]
LGTM. Thanks for the changes. Don't forget to do a local test run before re-creating the production resources.

**Comment by YifeiZhuang** [comment]
Runs successfully after creating the new url-map resources in prod. Merging.
https://fusion2.corp.google.com/invocations/bb8b643d-2d4f-4059-937f-0e2108411cc7/targets

**Pull request merged**